### PR TITLE
Deprecate unused tasks and simplify check_resource

### PR DESCRIPTION
**Pull request by tonyskapunk**
##### SUMMARY

The check_resource used to take actions for uncordoning nodes that is not in used in a long time. Most of the code was to check things but not to act on them, simplifying the role to only wait for the state of certain kind or resources.

Deprecating the use of oc for tasks that can be performed through the k8s module.
Removing options not in use.

##### ISSUE TYPE

- Docs Fix AND other nominal changes

##### Tests

- [x] TestDallas: ocp-4.17-vanilla example-cnf - 
  - [x] ocp4.17-vanilla -  https://www.distributed-ci.io/jobs/dd016668-cc08-467b-9303-7d9c218e00fb
  - [x] example-cnf - https://www.distributed-ci.io/jobs/c3f1a35e-cf42-4e06-bfc8-df083eedefe8
- [x] TestDallasWorkload: preflight-green tnf-test-cnf-green
  - [x] preflight - https://www.distributed-ci.io/jobs/09d23696-0fae-4c9a-bce7-015ede47292f
  - [x] tnf-test-cnf-green - https://www.distributed-ci.io/jobs/bee2bd45-91b0-411f-aa43-09fcef826d09
- [x] TestBos2: virt - https://www.distributed-ci.io/jobs/3b6e4396-2d9a-4e55-bb65-e979ffb1d485
- [x] TestBos2Sno: sno - https://www.distributed-ci.io/jobs/15f0388a-66ea-4fb3-afce-5685592a457f

---

Test-hints: no-check

### CODE REVIEW

File: roles/check_resource/tasks/main.yml [hunk 1]
```diff
@@ -1,24 +1,17 @@
...
+  ansible.builtin.set_fact:
+    cr_kind: "{{ resource_to_check }}"
+  when: resource_to_check is defined
 
-    # Check SriovNetworkNodeState application
-    - name: Check SriovNetworkNodeState application
-      ansible.builtin.include_tasks:
-        file: wait-sriov.yml
-      when: resource_to_check == "SriovNetworkNodeState"
+- name: Check MachineCconfigPool
```
Comment (betoredhat) [inline roles/check_resource/tasks/main.yml:7]: - name: Check MachineConfigPool

**Comment by tonyskapunk** [comment]
retest

**Comment by tonyskapunk** [comment]
retest

**Comment by tonyskapunk** [comment]
retest

**Comment by tonyskapunk** [comment]
retest

**Review by manurodriguez — APPROVED** [review]
LGTM

**Comment by tonyskapunk** [comment]
retest

**Review by betoredhat — APPROVED** [review]
LGTM, thanks

**Review by manurodriguez — APPROVED** [review]
Thanks

**Comment by tonyskapunk** [comment]
> The change does not affect the role usage of the agents. But I think you are already considering updating them separately.

Correct, the extra args are simply ignored, but I'll remove them from the places where this role is used.

**Pull request merged**